src/models/jObject.ts

```diff
@@ -0,0 +1,319 @@
+        attribute?: (value: string) => boolean
+    }): string {
+        let result = "";
+        const content = this.value;
+        let lineBreak = "\n";
+
+        for (let i = 0; i < identation; i++) {
+            lineBreak += " ";
+        }
+
+        switch (this.type) {
+            case "document":
+                this.children.forEach(child => {
+                    result += child.toFormattedXml(0, escapeCallbacks) + "\n";
+                });
+                break;
+
+            case "element":
+            case "template":
+                const tag = this.join([this.ns, this.name], ":");
+
+                result += `${lineBreak}<${tag}`;
+
+                this.attributes.forEach(attribute => {
+                    let value = attribute.value.toString();
+                    value = escapeCallbacks && escapeCallbacks.attribute && !escapeCallbacks.attribute(value) ? value : XmlUtil.encode(value);
+                    result += ` ${this.join([attribute.ns, attribute.name], ":")}="${value}"`;
+                });
+
+                if (this.children.length > 0) {
+                    result += `>`;
+
+                    this.children.forEach(child => {
+                        result += child.toFormattedXml(identation + 4, escapeCallbacks);
+                    });
+
+                    result += `${lineBreak}</${tag}>`;
+                } else if (content) {
+                    result += `>${content}</${tag}>`;
+                } else {
+                    result += ` />`;
+                }
+                break;
+
+            case "question":
+                result += this.value;
+                break;
+
+            case "comment":
+                result += `${lineBreak}<!--${content}-->`;
+                break;
+
+            case "cdata":
+                result += `<![CDATA[${content}]]>`;
+                break;
+
+            case "text":
+                if (content) {
+                    result += content;
+                }
+                break;
+
+            default:
+                throw new Error(`Unknown element type ${this.type}.`);
+        }
+
+        return result;
+    }
+
+    public toXml(escapeCallbacks?: { attribute?: (value: string) => boolean }): string {
+        return this.toFormattedXml(0, escapeCallbacks);
+    }
+
+    public innerXml(): string {
+        return this.children.map(x => x.toFormattedXml()).join();
+    }
+
+    public getAttribute(attributeName: string): string {
+        const attribute = this.attributes.find(x => x.name === attributeName);
+
+        if (attribute && attribute.value) {
+            return attribute.value;
+        }
+
+        return undefined;
+    }
+
+    public getAttributeAsNumber(attributeName: string): number {
+        const value = this.getAttribute(attributeName);
+
+        try {
+            return parseInt(value);
+        }
+        catch (error) {
+            return undefined;
+        }
```

Comment:
this block can be simplified as
return +value || undefined;

Reply:
Done